src/app-admin.js

```diff
@@ -0,0 +1,491 @@
+    return;
+  }
+
+  const { cnt } =
+    getAccountDb().first(
+      'SELECT count(*) AS cnt FROM user_access WHERE user_access.file_id = ? and user_access.user_id = ?',
+      [userAccess.fileId, userAccess.userId],
+    ) || {};
+
+  if (cnt > 0) {
+    sendErrorResponse(
+      res,
+      400,
+      'user-already-have-access',
+      'User already have access',
+    );
+    return;
+  }
+
+  getAccountDb().mutate(
+    'INSERT INTO user_access (user_id, file_id) VALUES (?, ?)',
+    [userAccess.userId, userAccess.fileId],
+  );
+
+  res.status(200).send({ status: 'ok', data: {} });
+});
+
+app.post('/access/delete-all', (req, res) => {
+  const fileId = req.query.fileId;
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  if (!checkFilePermission(fileId, session.user_id, res)) return;
+
+  const ids = req.body.ids;
+  let totalDeleted = 0;
+  ids.forEach((item) => {
+    const accessDeleted = getAccountDb().mutate(
+      'DELETE FROM user_access WHERE user_id = ?',
+      [item],
+    ).changes;
+    totalDeleted += accessDeleted;
+  });
+
+  if (ids.length === totalDeleted) {
+    res
+      .status(200)
+      .send({ status: 'ok', data: { someDeletionsFailed: false } });
+  } else {
+    sendErrorResponse(res, 400, 'not-all-deleted', '');
+  }
+});
+
+app.get('/access/users', async (req, res) => {
+  const fileId = req.query.fileId;
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  if (!checkFilePermission(fileId, session.user_id, res)) return;
+
+  const users = getAccountDb().all(
+    `SELECT users.id as userId, user_name as userName, display_name as displayName, 
+            CASE WHEN user_access.file_id IS NULL THEN 0 ELSE 1 END as haveAccess,
+            CASE WHEN files.id IS NULL THEN 0 ELSE 1 END as owner
+      FROM users
+      LEFT JOIN user_access ON user_access.file_id = ? and user_access.user_id = users.id
+      LEFT JOIN files ON files.id = ? and files.owner = users.id
+      WHERE users.enabled = 1 AND users.user_name <> ''`,
+    [fileId, fileId],
+  );
+  res.json(users);
+});
+
+app.post('/access/get-bulk', (req, res) => {
+  const fileIds = req.body || {};
+  const accessMap = new Map();
+
+  fileIds.forEach((fileId) => {
+    const userAccess = getAccountDb().all(
+      `SELECT user_access.file_id as fileId, user_access.user_id as userId, users.display_name as displayName, users.user_name as userName
+       FROM users
+       JOIN user_access ON user_access.user_id = users.id
+       WHERE user_access.file_id = ?
+       UNION
+       SELECT files.id, users.id, users.display_name, users.user_name
+       FROM users
+       JOIN files ON files.owner = users.id
+       WHERE files.id = ?`,
+      [fileId, fileId],
+    );
+    accessMap.set(fileId, userAccess);
+  });
+
+  res.status(200).send({ status: 'ok', data: Array.from(accessMap.entries()) });
+});
+
+app.get('/access/check-access', (req, res) => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏do we need this endpoint? Could we instead use the response of the initial sync for this?

i.e. - one of the endpoints returns a list of files for the current user. Then the user clicks on the file to access it. This triggers the download file endpoint. Can we use the "download file endpoint" to check access instead of having a dedicated endpoint for this same operation?

Reply:
i will implement this